it/controllers/benefits/reimbursed/NonTaxableCostsBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,528 @@
+              insertCyaData(employmentUserData(hasPriorBenefits = true, cyaModel("name", hmrc = true,
+                Some(benefits(Some(fullReimbursedCostsVouchersAndNonCashModel.copy(expenses = Some(22.00))))))), userRequest)
+              urlGet(nonTaxableCostsBenefitsAmountPageUrl(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            s"has an OK($OK) status" in {
+              result.status shouldBe OK
+            }
+
+            implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+            titleCheck(user.specificExpectedResults.get.expectedTitle)
+            h1Check(user.specificExpectedResults.get.expectedHeading)
+            captionCheck(expectedCaption, captionSelector)
+            textOnPageCheck(ifItWasNotText(amount), ifItWasNotTextSelector)
+            textOnPageCheck(enterTotalText, enterTotalSelector)
+            textOnPageCheck(expectedHintText, hintTextSelector)
+            inputFieldCheck(amountFieldName, amountFieldSelector)
+            inputFieldValueCheck("", amountFieldSelector)
+            buttonCheck(continueButtonText, continueButtonSelector)
+            formPostLinkCheck(formPostLink, formSelector)
+
+            welshToggleCheck(user.isWelsh)
+
+          }
+
+          "expenses is None" which {
+
+            implicit lazy val result: WSResponse = {
+              authoriseAgentOrIndividual(user.isAgent)
+              dropEmploymentDB()
+              userDataStub(userData(fullEmploymentsModel()), nino, taxYearEOY)
+              insertCyaData(employmentUserData(hasPriorBenefits = false, cyaModel("name", hmrc = true,
+                Some(benefits(Some(fullReimbursedCostsVouchersAndNonCashModel.copy(expenses = None)))))), userRequest)
+              urlGet(nonTaxableCostsBenefitsAmountPageUrl(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+            }
+
+            s"has an OK($OK) status" in {
+              result.status shouldBe OK
+            }
+
+            implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+            titleCheck(user.specificExpectedResults.get.expectedTitle)
+            h1Check(user.specificExpectedResults.get.expectedHeading)
+            captionCheck(expectedCaption, captionSelector)
+            textOnPageCheck(enterTotalText, enterTotalSelector)
+            elementNotOnPageCheck(ifItWasNotTextSelector)
+            textOnPageCheck(expectedHintText, hintTextSelector)
+            inputFieldCheck(amountFieldName, amountFieldSelector)
+            inputFieldValueCheck("", amountFieldSelector)
+            buttonCheck(continueButtonText, continueButtonSelector)
+            formPostLinkCheck(formPostLink, formSelector)
+
+            welshToggleCheck(user.isWelsh)
+
+          }
+        }
+
+        "render the non-taxable costs benefits amount page with a pre-filled amount field" when {
```

Comment:
its probably good to have a scenario where there is no prior value, so that you can check the "ifItWasNotText" is not displayed on the page.

Reply:
done. There's a check for element not on page in one of the tests